File: runtimes/web/src/engine/background/BackgroundAttachmentConfig.ts
// BackgroundAttachmentConfig.ts — typed record for CSS `background-attachment`.
// IR shape: array of { type: 'scroll'|'fixed'|'local' } (or bare strings).

// One entry per layer.
export interface BackgroundAttachmentLayer {
  css: string;                                                        // 'scroll'|'fixed'|'local'
}

// Config holder.
export interface BackgroundAttachmentConfig {
  layers: BackgroundAttachmentLayer[];                                // empty = unset
}

// IR property type.
export const BACKGROUND_ATTACHMENT_PROPERTY_TYPE = 'BackgroundAttachment' as const;
export type BackgroundAttachmentPropertyType = typeof BACKGROUND_ATTACHMENT_PROPERTY_TYPE;

File: runtimes/swiftui/Sources/StyleConverterRuntime/StyleEngine/background/BackgroundAttachmentConfig.swift
//
//  BackgroundAttachmentConfig.swift
//  StyleEngine/background — Phase 4.
//
//  CSS `background-attachment` (scroll / fixed / local). SwiftUI has no
//  equivalent — the background always scrolls with the view unless a
//  custom GeometryReader trick is used. Config is parsed for parity; the
//  applier is identity. Documented limitation.
//

import Foundation

// Per-layer attachment mode.
enum BackgroundAttachmentMode: Equatable {
    case scroll
    case fixed
    case local
}

struct BackgroundAttachmentConfig: Equatable {
    var layers: [BackgroundAttachmentMode] = []
    var hasAny: Bool { !layers.isEmpty }
}
